test/unit/test_amqpclient.js

```diff
@@ -153,136 +50,68 @@ describe('AMQPClient', function() {
+      var client = new Mock.Client();
+      expect(function () {
+        client.send();
+      }).to.throw(Error);
     });
 
-    it('should wait for capacity before sending', function() {
-      var c = new MockConnection();
-      var s = new MockSession(c);
-      var l = new MockLink(s, {
+    it('should wait for capacity before sending', function () {
+      var c = new Mock.Connection();
+      var s = new Mock.Session(c);
+      var l = new Mock.SenderLink(s, {
         name: 'queue_TX',
         isSender: true,
         capacity: 0
       });
 
       s._addMockLink(l);
-      var client = new MakeMockClient(c, s);
+      var client = new Mock.Client(c, s);
       var queue = 'queue';
-      var called = { open: 0, begin: 0, attachLink: 0, canSend: 0, sendMessage: 0 };
-      c.on('open-called', function(_c, _addr, _sasl) {
+      var called = {open: 0, begin: 0, attachLink: 0, canSend: 0, sendMessage: 0};
+      c.on('open-called', function (_c, _addr, _sasl) {
         expect(_addr).to.eql(u.parseAddress(mock_uri));
         expect(_sasl).to.not.exist;
         called.open++;
         _c.emit(Connection.Connected, _c);
       });
 
-      s.on('begin-called', function(_s, _policy) {
+      s.on('begin-called', function (_s, _policy) {
         called.begin++;
         _s.emit(Session.Mapped, _s);
       });
 
-      s.on('attachLink-called', function(_s, _policy, _l) {
+      s.on('attachLink-called', function (_s, _policy, _l) {
         called.attachLink++;
-        expect(client._pendingSends[_l.name]).to.not.be.empty;
-        expect(_policy.options.target).to.eql({ address: queue });
+        expect(_policy.options.target).to.eql({address: queue});
         expect(_policy.options.role).to.eql(constants.linkRole.sender);
-        process.nextTick(function() {
-          _l.capacity = 100;
-          _l.emit(Link.CreditChange, _l);
-        });
 
         _s.emit(Session.LinkAttached, _l);
       });
 
-      l.on('canSend-called', function() {
+      l.on('canSend-called', function () {
         called.canSend++;
       });
 
-      l.on('sendMessage-called', function(_l, id, msg, opts) {
+      l.on('sendMessage-called', function (_l, id, msg, opts) {
         called.sendMessage++;
-        process.nextTick(function() {
-          s.emit(Session.DispositionReceived, {
-            settled: true,
-            state: { },
-            first: id,
-            last: null
-          });
-        });
       });
 
       return client.connect(mock_uri)
-        .then(function() {
-          return client.send({ my: 'message' }, queue);
+        .then(function () {
+          return client.createSender(queue);
         })
-        .then(function() {
+        .then(function (sender) {
+          return Promise.all([
+            sender.send({ my: 'message' }),
+            process.nextTick(function() {
+              sender.capacity = 100;
+              sender._dispatchPendingSends();
```

Comment:
chalking this up to "I'm about to make a sender_link.test.js, so let's worry about it there"